sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1328,6 +1332,155 @@ void captureSessionToken(RxDocumentServiceRequest request, RxDocumentServiceResp
+
+
+                                     //group the itemKeyList based on partitionKeyRangeId
+                                     if (partitionRangeItemKeyMap.get(range) == null) {
+                                         List<Pair<String, PartitionKey>> list = new ArrayList<>();
+                                         list.add(stringPartitionKeyPair);
+                                         partitionRangeItemKeyMap.put(range, list);
+                                     } else {
+                                         List<Pair<String, PartitionKey>> pairs = partitionRangeItemKeyMap.get(range);
+                                         pairs.add(stringPartitionKeyPair);
+                                         partitionRangeItemKeyMap.put(range, pairs);
+                                     }
+
+                                 });
+
+                             Set<PartitionKeyRange> partitionKeyRanges = partitionRangeItemKeyMap.keySet();
+                             List<PartitionKeyRange> ranges = new ArrayList<>();
+                             ranges.addAll(partitionKeyRanges);
+
+                             //Create the range query map that contains the query to be run for that partitionkeyrange
+                             Map<PartitionKeyRange, String> rangeQueryMap;
+                             rangeQueryMap = getRangeQueryMap(partitionRangeItemKeyMap, collection.getPartitionKey());
+
+                             String sqlQuery = "this is dummy and only used for creating " +
+                                                   "ParallelDocumentQueryExecutioncontext, but not used";
+
+                             return createReadManyQuery(collectionLink, new SqlQuerySpec(sqlQuery), new FeedOptions(), Document.class,
+                                                        ResourceType.Document, collection, rangeQueryMap)
+                                        .collectList() // aggregating the result construct a FeedResponse and aggregate RUs.
+                                        .map(feedList -> {
+                                            List<Document> finalList = new ArrayList<>();
+                                            HashMap<String, String> headers = new HashMap<>();
+                                            double requestCharge = 0;
+                                            for (FeedResponse<Document> page : feedList) {
+                                                requestCharge += page.getRequestCharge();
+                                                finalList.addAll(page.getResults());
+                                            }
+                                            headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double.toString(requestCharge));
+                                            FeedResponse<Document> frp = BridgeInternal.createFeedResponse(finalList, headers);
```

Comment:
Split handling: Do query layer take care?

Reply:
`DocumentProducer` internally/recursively will handle split, if DocumentProducers correctly wired up no additional work for split proofing should be needed.